taar/recommenders/recommendation_manager.py

```diff
@@ -38,4 +38,11 @@ def recommend(self, client_id, limit):
         # Compute the recommendation.
         for r in self.recommenders:
             if r.can_recommend(client_info):
-                return r.recommend(client_info, limit)
+                logger.info("Using '{}' for client '{}'"
+                            .format(r.__class__.__name__, client_id))
+
+                recommendations = r.recommend(client_info, limit)
+                if len(recommendations) == 0:
+                    logger.info("No recommendations for client '{}'".format(client_id))
```

Comment:
Let's use the `extra` argument here as well and also record the recommender that returned zero results.